### PR TITLE
Fix parallel execution safety for temporary files

### PR DESCRIPTION
**Pull request by takahirom**
# What
Fixed temporary file creation to prevent collisions when multiple Claude Code instances run simultaneously by adding process ID to mktemp patterns.

# Why
Claude Code can run in parallel, causing temporary file name collisions with hardcoded paths. The mktemp pattern needed process-specific prefixes to ensure unique file names across concurrent sessions.

Changed from TMPDIR-based paths to mktemp -t with process ID ($$) for guaranteed uniqueness.

## Summary by CodeRabbit

- Bug Fixes
  - Improved temporary file handling to include process identifiers and safer cleanup, reducing filename collisions and handling edge-case names reliably.

- Tests
  - Updated performance tests to use the same robust temporary file strategy and safer removal, improving test stability and consistent setup/teardown.

### CODE REVIEW

File: .claude/scripts/common-config.sh [hunk 1]
```diff
@@ -279,7 +279,7 @@ count_attempts_since_last_reset_point() {
     # Stop hook messages appear in tool_use commands within assistant messages
     local attempt_count=0
     local temp_transcript
-    temp_transcript="$(mktemp "${TMPDIR:-/tmp}/filtered_transcript.XXXXXX.jsonl")"
+    temp_transcript="$(mktemp -t filtered_transcript.$$.XXXXXX)" && temp_transcript="${temp_transcript}.jsonl"
     trap 'rm -f "$temp_transcript"' RETURN
```
[the remark this answers is not in the document]
Comment (takahirom) [reply .claude/scripts/common-config.sh]: This is function scope, in that case, doesn't it suitable to use RETURN?
